designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ScenarioLabelsApiHttpServiceSecuritySpec.scala

```diff
@@ -0,0 +1,290 @@
+          .Then()
+          .statusCode(200)
+          .equalsJsonBody(
+            s"""{
+               |  "labels": ["tag1", "tag2", "tag3", "tag4"]
+               |}""".stripMargin
+          )
+      }
+    }
+    "not authenticated should" - {
+      "forbid access" in {
+        given()
+          .when()
+          .basicAuthUnknownUser()
+          .get(s"$nuDesignerHttpAddress/api/scenarioLabels")
+          .Then()
+          .statusCode(401)
+      }
+    }
+    "no credentials were passed should" - {
+      "authenticate as anonymous and return no labels for no write access" in {
+        val scenario1 = exampleScenario("s1")
+        val scenario2 = exampleScenario("s2")
+
+        given()
+          .applicationState {
+            createSavedScenario(scenario1, category = Category1)
+            updateScenarioLabels(scenario1, List("tag2", "tag3"))
+            createSavedScenario(scenario2, category = Category1)
+            updateScenarioLabels(scenario2, List("tag1", "tag4"))
+          }
+          .when()
+          .noAuth()
+          .get(s"$nuDesignerHttpAddress/api/scenarioLabels")
+          .Then()
+          .statusCode(200)
+          .equalsJsonBody(
+            """{
+              |  "labels": []
+              |}""".stripMargin
+          )
+      }
+    }
+    "impersonating user has permission to impersonate should" - {
+      "return all scenario labels" in {
+        val scenario1 = exampleScenario("s1")
+        val scenario2 = exampleScenario("s2")
+
+        given()
+          .applicationState {
+            createSavedScenario(scenario1, category = Category1)
+            updateScenarioLabels(scenario1, List("tag2", "tag3"))
+            createSavedScenario(scenario2, category = Category1)
+            updateScenarioLabels(scenario2, List("tag1", "tag4"))
+          }
+          .when()
+          .basicAuthAllPermUser()
+          .impersonateLimitedWriterUser()
+          .get(s"$nuDesignerHttpAddress/api/scenarioLabels")
+          .Then()
+          .statusCode(200)
+          .equalsJsonBody(
+            s"""{
+               |  "labels": ["tag1", "tag2", "tag3", "tag4"]
+               |}""".stripMargin
+          )
+      }
+    }
+    "impersonating user does not have permission to impersonate should" - {
+      "forbid access" in {
+        given()
+          .when()
+          .basicAuthWriter()
+          .impersonateLimitedWriterUser()
+          .get(s"$nuDesignerHttpAddress/api/scenarioLabels")
+          .Then()
+          .statusCode(403)
+          .equalsPlainBody("The supplied authentication is not authorized to impersonate")
+      }
+    }
+  }
+
+  "The scenario labels validation endpoint when" - {
+    "authenticated should" - {
+      "validate scenario labels" in {
+        given()
+          .when()
+          .basicAuthAdmin()
+          .body(
+            s"""
+               |{
+               |  "labels": ["tag100", "tag2"]
```

Comment:
Could you also add validation for duplicated labels? Right now if you try to have 2 of the same label I think the FE will just not make the request. WDYT about FE sending all the current labels and for `["tag2", "tag2"]` it would return error with info that labels cannot be duplicated / have to be unique?